addon/authenticators/oidc.js

```diff
@@ -124,6 +125,8 @@ export default BaseAuthenticator.extend({
         retryCount < amountOfRetries - 1
       ) {
         later(this, this._refresh, refresh_token, retryCount + 1, retryTimeout);
+        // Return old token as we couldn't refresh yet
+        return this._currentTokenData;
```

Comment:
Would it not make more sense to use the current `sessionData` instead of adding another state to the addon?

Reply:
Agree

Reply:
I think it should not return anything at all. Instead of returning old data, we should focus on only triggering `sessionDataChanged` if something actually changed.

Reply:
Do we have access to the current sessionData here? If so, I agree this would be better